Change Gtk.VBox & Gtk.HBox to Gtk.Box

tristan957 opened the pull request:
With the upcoming Gtk 4 changes, Gtk.VBox and Gtk.HBox have been deprecated in favor of Gtk.Box with an orientation.

tristan957 committed:
Change Gtk.VBox & Gtk.HBox to Gtk.Box

With the upcoming Gtk 4 changes, Gtk.VBox and Gtk.HBox have been deprecated in favor of Gtk.Box with an orientation.

tristan957 commented:
FYI I wasn't able to test my changes, but I tried my best to double check stuff for spelling. Let me know if I should do something different stylistically, or fix a bug.

ikeydoherty commented on os_installer2/pages/users.py:
```diff
@@ -20,11 +20,13 @@
 
     def __init__(self, user):
-        Gtk.VBox.__init__(self)
+        Gtk.Box.__init__(self)
```
`Gtk.Box.__init__(self, orientation=Gtk.Orientation.VERTICAL, spacing=0)` should also work?

tristan957 replied:
Yea I was trying to be consistent with what I saw in the code. I'll make the change

ikeydoherty replied:
Yeah my older code, especially Python, can leave a lot to be desired :D

ikeydoherty commented on os_installer2/pages/users.py:
```diff
@@ -287,7 +289,8 @@ def __init__(self):
         self.pack_start(self.stack, True, True, 0)
 
-        main_page = Gtk.VBox()
+        main_page = Gtk.Box()
```
I'd rather they all be consistent and not use separate property calls :)

ikeydoherty commented on os_installer2/mainwindow.py:
```diff
@@ -167,7 +167,7 @@ def __init__(self, app):
 
         # Main "install" page
-        self.installer_page = Gtk.VBox(0)
+        self.installer_page = Gtk.Box(Gtk.Orientation.VERTICAL, 0)
```
pygi can be buggy, probably safer to explicitly use `Gtk.Box.new(Gtk.Orientation.VERTICAL, 0)`

tristan957 replied:
My bad. I'll fix that tonight.

ikeydoherty replied:
S'all good, appreciate the fixes!

tristan957 committed:
Fix suggestions per Ikey

Use of Gtk.Box.new() as well as stating orientations in Gtk.Box.__init_() contructors

tristan957 committed:
Also explicitly state spacing=0 in __init__

tristan957 commented:
Alright @ikeydoherty. Couple weeks later, but I think I got what you wanted me to do. I have explicitly stated `orientation=Gtk.Orientation.[ORIENTATION]` and `spacing=0` in `Gtk.Box.__init__()` as well as changing all `Gtk.Box()` to `Gtk.Box.new()`. The only thing you may have to watch out for is if I put a `spacing=0` where it should have been something else. I might've missed something like that.

ikeydoherty commented:
Looks legit - thanks bud, if it does crash ill fix it up dont worry, it'd be during startup anyway. Need to get this installer modernised :D

tristan957 commented:
No problemo!
